### ISSUE TITLE
How to trigger action on route change ?

### ISSUE THREAD

**Issue by Grustamli** [solidjs/solid-start#808]
I have a global store which holds list of messages. I'd like to clear the messages when route changes.

How can I achieve this? I checked `createRouteAction` but it seems it is not automatically called when navigating.

**Comment by boehs** [comment]
That's not what actions are for. Use the createData functions

**Comment by Grustamli** [comment]
Do you mean `createRouteData` function, because I didn't find createDats in the docs?

**Comment by boehs** [comment]
> Do you mean `createRouteData` function, because I didn't find createDats in the docs?

yeah. there are variations like createRouteData, createServerData, ect.

**Comment by boehs** [comment]
actions are for data mutations

**Comment by peerreynders** [comment]
@Grustamli have a look at [`useIsRouting()`](https://start.solidjs.com/api/useIsRouting)

If necessary you can pass the signal from `root.tsx` into [context](https://www.solidjs.com/docs/latest/api#createcontext) where you manipulate the store. For an example [see this](https://github.com/peerreynders/solid-start-todomvc-kcd-v2/blob/main/src/root.tsx) or [this](https://github.com/peerreynders/solid-start-demo-login/blob/9005e4a9394fc625a0f7876d288e5c0829f3ba41/src/components/user-context.tsx#L39-L73).
